### DISCUSSION TITLE
Pastas to back-calculate and predict other hydrological variables

### DISCUSSION THREAD

**Discussion by LucaPiciullo** [pastas/pastas#670] [Q&A]
HI,
I am wondering if it would be possible to use pastas to back-calculate and predictid the trend of variables different that groundwater level. Specifically I am thinking about water content and pore water pressure. 
We have a large dataset available.

best
LPi

**Comment by dbrakenhoff** [comment]
Hi @LucaPiciullo,

I'm not entirely sure what you mean by back-calculate, but it's certainly possible to try to build a model that simulates water content or pore-water-pressure, given certain forcing stresses. Be aware that Pastas was mainly tested on groundwater levels, so not all the default choices/settings might be optimal for different variables.

As for predictions, once you have a trained model, you can use the estimated parameters to simulate the response of your simulated variable given new forcing stresses. 
```python
# build model
ml = ps.Model(your_time_series)
sm = ps.StressModel(stress, ...)  # make sure your stress contains data for both the calibration period and prediction period
ml.add_stressmodel(sm)
ml.solve(tmin=tmin, tmax=tmax)  # set calibration period

prediction = ml.simulate(tmin=tmax)  # prediction
```
See also the [notebook on uncertainty quantification](https://pastas.readthedocs.io/en/latest/examples/uncertainty.html) for more tips on how to get an ensemble of predictions given model parameter uncertainty.

Hope that answers your question somewhat?

**Reply by LucaPiciullo** [reply]
Thanks @dbrakenhoff 
about back-calculation I simply mean the time serie analysis simulation. So to use forcing stresses (i,e, relevant varaibles) that influence the water content (or pore water pressure) to simulate the trend and then make predictions. 
Basically the variables to predict groudwater level and water content (or pore water pressure) are somehow the same.

**Comment by OnnoEbbens** [comment]
Pastas has been applied to soil moisture time series before. I don't know too much about it but there is a paper with more information: https://www.sciencedirect.com/science/article/pii/S1364815220300876

**Reply by LucaPiciullo** [reply]
thank you very much for this reference
